### PR TITLE
Fix hack/dev-push-hyperkube.sh

### PR DESCRIPTION
**Pull request by colemickens**
**What this PR does / why we need it**: This fixes `hack/dev-push-hyperkube.sh`. It simplifies/standardizes its usage and fixes it for the changes to the build system.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34415)

### CODE REVIEW

**Comment by colemickens** [comment]
Added another commit that speeds up the hyperkube image build by adding `/hyperkube` at the end of the Dockerfile.

File: cluster/images/hyperkube/Dockerfile [hunk 1]
```diff
@@ -78,3 +75,7 @@ RUN ln -s /hyperkube /apiserver \
  && ln -s /hyperkube /kubelet \
  && ln -s /hyperkube /proxy \
  && ln -s /hyperkube /scheduler
+
+# Copy the hyperkube binary
+COPY hyperkube /hyperkube
```
Comment (luxas) [inline cluster/images/hyperkube/Dockerfile]: I do not think this can come before the `ln -s` lines above that are referencing this file

**Comment by colemickens** [comment]
It can. I tested images built with these changes. It's fine to create the
links and then place the file.

**Comment by lavalamp** [comment]
lgtm

**Comment by colemickens** [comment]
It significantly cuts down on rebuild time. Adding it in the beginning
invalidates the entire cache and forces you to go through all of apt-get
commands again.

File: cluster/images/hyperkube/Dockerfile (hunk 1 above)
Comment (colemickens) [inline cluster/images/hyperkube/Dockerfile]: (Comment didn't get added here correctly...) It significantly cuts down on rebuild time. Adding it in the beginning
invalidates the entire cache and forces you to go through all of apt-get
commands again.

**Comment by colemickens** [comment]
@k8s-bot gci gce e2e test this

File: cluster/images/hyperkube/Dockerfile (hunk 1 above)
Comment (colemickens) [inline cluster/images/hyperkube/Dockerfile]: I guess it was already after the apt commands, but still this exploits the cache slightly better.

**Comment by lavalamp** [comment]
ah, I see! very nice.

**Pull request merged**